extensions/omni_rest/src/_postgrest_function.sql

```diff
@@ -0,0 +1,95 @@
+create or replace function _postgrest_function_by_arguments (namespace text, function_name text, function_arguments text[])
+    returns regproc[]
+    language sql
+    stable
+    as $$
+    with arguments as (
+        select
+            oid,
+            array_agg(coalesce(name, '')) filter (where idx <= (pronargs - pronargdefaults)) as required_parameters,
+        array_agg(coalesce(name, '')) as parameters
+    from
+        pg_proc,
+        unnest(proargnames, proargtypes, proargmodes)
+        with ordinality as _ (name, type, mode, idx)
+    where
+        type is not null -- only input arguments
+    group by
+        oid
+        -- we can only work with named arguments
+    having
+        count(*) - count(name) = 0
+)
+select
+    array_agg(p.oid::regproc)
+from
+    pg_proc p
+    left join arguments a on a.oid = p.oid
+    left join pg_catalog.pg_namespace n on n.oid = p.pronamespace
+where
+    n.nspname = namespace
+    and p.proname = function_name
+    and coalesce(required_parameters, '{}') <@ function_arguments
+    and (function_arguments <@ coalesce(parameters, '{}')
+        or function_arguments = coalesce(parameters, '{}'))
+$$;
+
+create procedure _postgrest_function (request omni_httpd.http_request, function_reference inout regproc, namespace inout text, settings postgrest_settings default postgrest_settings ())
+language plpgsql
+as $$
+declare
+    saved_search_path text := current_setting('search_path');
+    function_name text;
+    function_arguments text[];
+    all_references regproc[];
+begin
+    if request.method = 'GET' then
+        namespace := omni_http.http_header_get (request.headers, 'accept-profile');
+    end if;
+    if request.method = 'POST' then
```

Comment:
super minor: `else` or `case` to avoid two checks in a row when first matched?

Reply:
indeed, will change that.